com.unity.render-pipelines.core/Editor/Debugging/DebugUIDrawer.Builtins.cs

```diff
@@ -654,6 +654,80 @@ public override bool OnGUI(DebugUI.Widget widget, DebugState state)
         }
     }
 
+    /// <summary>
+    /// Builtin Drawer for Object Debug Items.
+    /// </summary>
+    [DebugUIDrawer(typeof(DebugUI.ObjectField))]
+    public sealed class DebugUIDrawerObjectField : DebugUIDrawer
+    {
+        /// <summary>
+        /// OnGUI implementation for Object DebugUIDrawer.
+        /// </summary>
+        /// <param name="widget">DebugUI Widget.</param>
+        /// <param name="state">Debug State associated with the Debug Item.</param>
+        /// <returns>The state of the widget.</returns>
+        public override bool OnGUI(DebugUI.Widget widget, DebugState state)
+        {
+            var w = Cast<DebugUI.ObjectField>(widget);
+
+            EditorGUI.BeginChangeCheck();
+
+            var rect = PrepareControlRect();
+            var value = EditorGUI.ObjectField(rect, EditorGUIUtility.TrTextContent(w.displayName, w.tooltip), w.GetValue(), w.type, true);
+
+            if (EditorGUI.EndChangeCheck())
+                Apply(w, state, value);
+
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Builtin Drawer for Object Debug Items.
+    /// </summary>
+    [DebugUIDrawer(typeof(DebugUI.ObjectListField))]
+    public sealed class DebugUIDrawerObjectListField : DebugUIDrawer
+    {
+        /// <summary>
+        /// OnGUI implementation for ObjectList DebugUIDrawer.
+        /// </summary>
+        /// <param name="widget">DebugUI Widget.</param>
+        /// <param name="state">Debug State associated with the Debug Item.</param>
+        /// <returns>The state of the widget.</returns>
+        public override bool OnGUI(DebugUI.Widget widget, DebugState state)
+        {
+            var w = Cast<DebugUI.ObjectListField>(widget);
+            var objects = w.GetValue();
+
+            float height = Math.Max(objects != null ? objects.Length : 0, 1) * DebugWindow.Styles.singleRowHeight;
+            var rect = PrepareControlRect(height);
+
+            rect = EditorGUI.PrefixLabel(rect, EditorGUIUtility.TrTextContent(widget.displayName));
+
+            EditorGUI.BeginChangeCheck();
+            DoObjectList(rect, w, objects);
+            if (EditorGUI.EndChangeCheck())
+                Apply(w, state, objects);
+
+            return true;
+        }
+
+        internal static void DoObjectList(Rect rect, DebugUI.ObjectListField widget, UnityEngine.Object[] objects)
+        {
+            if (objects != null && objects.Length != 0)
+            {
+                rect.height = EditorGUIUtility.singleLineHeight;
+                for (int i = 0; i < objects.Length; i++)
+                {
+                    objects[i] = EditorGUI.ObjectField(rect, GUIContent.none, objects[i], widget.type, true);
+                    rect.y += DebugWindow.Styles.singleRowHeight;
+                }
+            }
+            else
+                EditorGUI.LabelField(rect, GUIContent.none, EditorGUIUtility.TrTextContent("Empty"));
```

Comment:
Invert condition to avoid nesting and early return.
```suggestion
            if (objects == null || objects.Length == 0)
            {
             EditorGUI.LabelField(rect, GUIContent.none, EditorGUIUtility.TrTextContent("Empty"));
             return;
             }

                rect.height = EditorGUIUtility.singleLineHeight;
                for (int i = 0; i < objects.Length; i++)
                {
                    objects[i] = EditorGUI.ObjectField(rect, GUIContent.none, objects[i], widget.type, true);
                    rect.y += DebugWindow.Styles.singleRowHeight;
                }
```

Reply:
I think we have the "Empty" `GuiContent on `CoreEditorStyles`

Reply:
the empty of CoreEditorStyles is really empty, so it doesn't work here